lib/SPIRV/libSPIRV/SPIRVDecorate.h

```diff
@@ -149,6 +150,11 @@ class SPIRVDecorateLinkageAttr:public SPIRVDecorate{
   SPIRVLinkageTypeKind getLinkageType() const {
     return (SPIRVLinkageTypeKind)Literals.back();
   }
+
+  void encode(std::ostream &O)const {
+    getEncoder(O) << Target << Dec << getLinkageName() << SPIRVTextOnly(" ")

```

Comment:
should only do this if SPIRVUseTextFormat is true, otherwise still need output binary format.

Also, need corresponding decode() function.